### PR TITLE
Add Gateway CRD

### PR DESCRIPTION
**Pull request by skitt**
This will be used to store and report the cluster-local engine
status.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: pkg/broker/rbac.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func NewBrokerAdminRole() *rbacv1.Role {
 			rbacv1.PolicyRule{
 				Verbs:     []string{"create", "get", "list", "watch", "patch", "update", "delete"},
 				APIGroups: []string{"submariner.io"},
-				Resources: []string{"clusters", "endpoints"},
+				Resources: []string{"clusters", "endpoints", "gateways"},
```
Comment (mangelajo) [inline pkg/broker/rbac.go:57]: Those rbacs are for the broker cluster, we should not handle gateways on the broker cluster. This admin role is for subctl over the broker.

File: pkg/broker/rbac.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func NewBrokerClusterRole() *rbacv1.Role {
...
```
Comment (mangelajo) [inline pkg/broker/rbac.go:84]: Those rbacs are for the broker cluster, we should not handle gateways on the broker cluster.

File: deploy/cluster_role.yaml [hunk 3]
```diff
@@ -21,7 +21,10 @@ rules:
   resources:
   - endpoints
   - clusters
+  - gateways
```
Comment (mangelajo) [inline deploy/cluster_role.yaml:24]: This cluster_role.yaml is only for globalnet I think. It may not need to change the permissions

Comment (skitt) [reply deploy/cluster_role.yaml:24]: OK, I’ll remove these.

**Pull request merged**